### DISCUSSION TITLE
Bind-mount: Host file system not same then jail files system

### DISCUSSION THREAD

**Discussion by PackElend** [Jip-Hop/jailmaker#50] [Q&A]
Hello,
I'm a bit confused here as `dir /mnt/SSD/jailmaker/jails/myjail/rootfs/home` does not list the same as `jlmkr exec myjail dir /home`.
I started the jail with 
```
startup=1
docker_compatible=1
gpu_passthrough_intel=0
gpu_passthrough_nvidia=0
systemd_nspawn_user_args=--network-macvlan=br8 --resolv-conf=bind-host --bind='/mnt/SSD/SysAdmin/tmp:/home'
# You generally will not need to change the options below
systemd_run_default_args=--property=KillMode=mixed --property=Type=notify --property=RestartForceExitStatus=133 --property=SuccessExitStatus=133 --property=Delegate=yes --property=TasksMax=infinity --collect --setenv=SYSTEMD_NSPAWN_LOCK=0
systemd_nspawn_default_args=--keep-unit --quiet --boot
```
so I would expect that I see `TN: '/mnt/SSD/SysAdmin/tmp`' in `Jail: '/home'` but when I browse the file system with either [filebrowser](https://github.com/filebrowser/filebrowser) or WinSCP I see the file created with [filebrowser](https://github.com/filebrowser/filebrowser)
<img src="https://github.com/Jip-Hop/jailmaker/assets/3542031/c3472a57-69f6-43ba-8dcf-e453df79fa03.png" width=20% height=20%> <img src="https://github.com/Jip-Hop/jailmaker/assets/3542031/5b782c98-8f8d-4a9e-9527-3369c870746d.png" width=20% height=20%>

```
root@truenas[~]# jlmkr exec myjail dir /home
neuer\ test.txt  ROHRKLASSEN\ ASME\ Anwendungen.pdf

dir /mnt/SSD/SysAdmin/tmp                
ROHRKLASSEN\ ASME\ Anwendungen.pdf  neuer\ test.txt

root@truenas[~]# dir /mnt/SSD/jailmaker/jails/myjail/rootfs/home
test_fs.txt
```

**Comment by Jip-Hop** [comment]
You have made /mnt/SSD/SysAdmin/tmp available at /home inside the jail. So files in the tmp dir will be available in the home dir from the perspective of the jail. Files that reside in /mnt/SSD/jailmaker/jails/myjail/rootfs/home will be 'hidden' from the perspective of the jail since you have mounted another directory over it (the tmp dir). Makes sense?

**Comment by PackElend** [comment]
It makes sense.
Only supprised that the jail is that well isolated, that the host does not see the mount into the jail (if my interpretation is right).

To see the files mounted into the jail with winSCP, I need to install an ftp or ssh server inside the jail isn't it?

**Reply by Jip-Hop** [reply]
I suppose so. But since files in the jail must also exist somewhere outside the jail I suppose you could just use the ssh server on the TrueNAS host to expose the desired /mnt/SSD/SysAdmin/tmp directory via sftp.

**Reply by PackElend** [reply]
That is true but sometimes it is helpful to see what the jail sees using a tool with a GUI
